catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -98,9 +122,77 @@ public void init() {
         "/cql",
         APPLICATION_JSON,
         (req, res) -> {
-          CqlRequest cqlRequest = mapper.readValue(util.safeGetBody(req), CqlRequest.class);
+          String transformerId = req.queryParams("transform");
+          String body = util.safeGetBody(req);
+          CqlRequest cqlRequest = mapper.readValue(body, CqlRequest.class);
           CqlQueryResponse cqlQueryResponse = executeCqlQuery(cqlRequest);
-          return mapper.toJson(cqlQueryResponse);
+
+          if (StringUtils.isEmpty(transformerId)) {
+            LOGGER.trace("Returning cql query response.");
+            return mapper.toJson(cqlQueryResponse);
+          }
+          for (ServiceReference<QueryResponseTransformer> queryResponseTransformer :
+              queryResponseTransformers) {
+
+            LOGGER.trace("Finding transformer to transform query response.");
+
+            String id = (String) queryResponseTransformer.getProperty("id");
+
+            if (!id.equals(transformerId)) {
+              continue;
+            }
+
+            BinaryContent content =
+                bundleContext
+                    .getService(queryResponseTransformer)
+                    .transform(this.queryResponse, new HashMap<>());
+
+            String mimeType = (String) queryResponseTransformer.getProperty("mime-type");
+            MimeTypes allTypes = MimeTypes.getDefaultMimeTypes();
+            String fileExt = allTypes.forName(mimeType).getExtension();
+
+            res.status(200);
```

Comment:
✏️ `Response.OK`

Reply:
Probably shouldn't be using jaxrs within a spark endpoint.

Reply:
I don't believe this is included in DDF anyway